src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtlDocumentTransformer.cs

```diff
@@ -348,19 +348,21 @@ public override void Transform(RavenEtlItem item, EtlStatsScope stats, EtlProces
             }
         }
 
-        private bool ShouldLoadTimeSeriesWithDoc(RavenEtlItem item, EtlProcessState state)
+        private bool ShouldLoadTimeSeriesWithDoc(Document doc, EtlProcessState state)
         {
             //If an Etag of time-series is lower then its document Etag then replication can send the time-series before the document.
             //In this situation Etl process will skip the time-series, mark it as skipped and will send all of it here with the document
-            return state.SkippedTimeSeriesDocs != null && state.SkippedTimeSeriesDocs.Remove(item.DocumentId);
+            return state.SkippedTimeSeriesDocs != null && state.SkippedTimeSeriesDocs.Remove(doc.Id);
         }
 
-        private bool ShouldFilterOutDeletion(RavenEtlItem item)
+        private bool ShouldFilterOutDeletion(Document doc, bool isDelete)
         {
             if (_script.HasDeleteDocumentsBehaviors)
             {
-                var collection = item.Collection ?? item.CollectionFromMetadata;
-                var documentId = item.DocumentId;
+                doc.TryGetMetadata(out var metadata);
+
+                var collection = metadata[Constants.Documents.Metadata.Collection].ToString();
+                var documentId = doc.Id;
```

Comment:
How does it work for tombstones? Note that previously we used:

https://github.com/ravendb/ravendb/blob/afbe5aa6af603914632c3b3291939106eb0afb68/src/Raven.Server/Documents/ETL/ExtractedItem.cs#L38

Reply:
outdated approach